storm-client/src/jvm/org/apache/storm/grouping/LoadAwareShuffleGrouping.java

```diff
@@ -85,6 +85,7 @@ public void prepare(WorkerTopologyContext context, GlobalStreamId stream, List<I
         sourceNodeInfo = new NodeInfo(context.getThisWorkerHost(), Sets.newHashSet((long) context.getThisWorkerPort()));
         taskToNodePort = context.getTaskToNodePort();
         this.targetTasks = targetTasks;
+        CAPACITY = targetTasks.size() == 1 ? 1 : targetTasks.size() * MAX_WEIGHT;
```

Comment:
Given that other logic still leverage max weight, will this work for targetTasks.size() == 1? If it works, It may be ideal to have test on this case to clear out wondering.

Reply:
`targetTasks.size() == 1`  is just for performance enhancement. Since there is only one target task, there is no need to compose a choice array and choose the target from the elements of the choice array; Set CAPACITY to 1 can avoid this. 
For the particular equation,
`int count = (int) ((indexAndWeights.weight / (double) weightSum) * CAPACITY);`
 because there is only one target, so `indexAndWeights.weight / (double) weightSum` must be 1 and count must be 1 if there is any load on that target task.

Reply:
OK we are ensuring it already.